src/Modules/Processing/CustomPlayer/CustomPlayer.cpp

```diff
@@ -34,12 +34,58 @@ void CustomPlayer::update() {
 }
 
 void CustomPlayer::exec() {
-  if (!field || !frame || !robot) {
+
+  if (!field || !frame || !robot)
+    return;
+  if (!frame->has_ball())
     return;
+  auto&& pos_ball = frame->ball();
+
+  QPointF targetPosition(1000, 4000);
+  QPointF initialPosition(2000, 0);
+  int tolerance = 150;
+  maxdis = 9000000;
+  if ((pos_ball.distTo(targetPosition)) > tolerance) {
```

Comment:
Colocar nomes descritivos para as condicionais facilita entender o que tá acontecendo:
```suggestion
  bool isBallOnTargetPosition = (pos_ball.distTo(targetPosition)) <= tolerance;
  if (!isBallOnTargetPosition) {
```

Reply:
Esse comentário também se aplica às outras condicionais